and-bible/app/src/main/java/net/bible/service/device/ProgressNotificationManager.kt

```diff
@@ -0,0 +1,150 @@
+package net.bible.service.device
+
+import android.app.*
+import android.content.Intent
+import android.os.Build
+import android.support.v4.app.NotificationCompat
+import android.support.v4.util.ArraySet
+import android.util.Log
+
+import net.bible.android.BibleApplication
+import net.bible.android.SharedConstants
+import net.bible.android.activity.R
+import net.bible.android.view.activity.download.ProgressStatus
+
+import org.apache.commons.lang3.StringUtils
+import org.crosswire.common.progress.JobManager
+import org.crosswire.common.progress.Progress
+import org.crosswire.common.progress.WorkEvent
+import org.crosswire.common.progress.WorkListener
+
+/**Show all Progress status
+ * see BibleDesktop JobsProgressBar for example use
+ *
+ * @author Martin Denham [mjdenham at gmail dot com]
+ * @see gnu.lgpl.License for license details.<br></br>
+ * The copyright to this program is held by it's author.
+ */
+
+// only one instance initialised at startup to monitor for JSword Progress events and
+// map them to Android Notifications
+
+class ProgressNotificationManager {
+    internal var progs: MutableSet<Progress> = ArraySet()
+    private var workListener: WorkListener? = null
+
+    // add it to the NotificationManager
+    lateinit private var notificationManager: NotificationManager
+
+    fun initialise() {
+        Log.i(TAG, "Initializing")
+        notificationManager = BibleApplication.getApplication().getSystemService(Application.NOTIFICATION_SERVICE) as NotificationManager
+        val app = BibleApplication.getApplication()
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val channel = NotificationChannel(DOWNLOAD_NOTIFICATIONS_CHANNEL,
```

Comment:
We should also make the channel name generic because it is not just for downloads

Reply:
Changed.